### PR TITLE
start working on ledger

### PR DESCRIPTION
**Pull request by upacyxou**
- start working on ledger

### CODE REVIEW

File: src/models/WorkLedger.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { SCERC721Derivative } from '@big-whale-labs/seal-cred-ledger-contract'
+
+export default interface EmailLedger {
+  [domain: string]: {
+    derivativeContract: SCERC721Derivative
```
Comment (backmeupplz) [inline src/models/WorkLedger.ts:5]: This is `SCEmailDerivative`

File: src/models/Proof.ts [hunk 2]
```diff
@@ -1,7 +1,6 @@
 import ProofResult from 'models/ProofResult'
 
 export default interface Proof {
-  contract: string
-  account: string
   result?: ProofResult
+  key: string
```
Comment (backmeupplz) [inline src/models/Proof.ts:5]: What's this `key`? Should this be `domain`?

File: src/components/badges/List.tsx [hunk 3]
```diff
@@ -29,32 +29,30 @@ const badgesList = classnames(
   gridTemplateColumns('grid-cols-1', 'lg:grid-cols-2')
 )
 function BadgeListSuspended() {
-  const { derivativeContracts = [] } = useSnapshot(SealCredStore)
+  const { erc721DerivativeContracts = [] } = useSnapshot(SealCredStore)
   const { contractsOwned } = useSnapshot(ContractsStore)
 
-  const ownedDerivativeContracts = derivativeContracts.filter(
+  const ownederc721DerivativeContracts = erc721DerivativeContracts.filter(
```
Comment (backmeupplz) [inline src/components/badges/List.tsx:35]: capitalize words

File: src/helpers/getLedgerRecord.ts [hunk 4]
```diff
@@ -2,6 +2,7 @@ import {
   ERC721__factory,
   SCERC721Derivative__factory,
 } from '@big-whale-labs/seal-cred-ledger-contract'
+
```
Comment (backmeupplz) [inline src/helpers/getLedgerRecord.ts:5]: I this an unnecessary change?

File: src/helpers/getLedger.ts [hunk 5]
```diff
@@ -17,5 +17,6 @@ export default async function (): Promise<Ledger> {
     },
     {}
   )
+  console.log('loaded erc721 ledger')
```
Comment (backmeupplz) [inline src/helpers/getLedger.ts:20]: Don't forget to remove it before sending for review

File: src/helpers/getLedger.ts [hunk 6]
```diff
@@ -1,10 +1,10 @@
+import { SealCredERC721Ledger } from '@big-whale-labs/seal-cred-ledger-contract'
```
Comment (backmeupplz) [inline src/helpers/getLedger.ts:1]: You should probably call this file `getERC721Ledger`

File: src/helpers/env.ts [hunk 7]
```diff
@@ -12,4 +12,5 @@ export default cleanEnv(import.meta.env, {
   VITE_ETH_RPC: str({ default: ETH_RPC }),
   VITE_SCLEDGER_CONTRACT_ADDRESS: str({ default: SCLEDGER_CONTRACT_ADDRESS }),
   VITE_VERIFY_URL: str(),
+  VITE_SCWPLEDGER_CONTRACT_ADDRESS: str(),
```
Comment (backmeupplz) [inline src/helpers/env.ts:15]: VITE_SC_EMAIL_LEDGER_CONTRACT_ADDRESS

File: src/helpers/env.ts [hunk 8]
```diff
@@ -12,4 +12,5 @@ export default cleanEnv(import.meta.env, {
   VITE_ETH_RPC: str({ default: ETH_RPC }),
   VITE_SCLEDGER_CONTRACT_ADDRESS: str({ default: SCLEDGER_CONTRACT_ADDRESS }),
```
Comment (backmeupplz) [inline src/helpers/env.ts:13]: VITE_SC_ERC721_LEDGER_CONTRACT_ADDRESS

File: .env.sample [hunk 9]
```diff
@@ -3,4 +3,5 @@ VITE_APP_NAME=Seal-Cred
 VITE_ETH_NETWORK=rinkeby
 VITE_ETH_RPC=https://eth.example.com/rpc
 VITE_SC_LEDGER_CONTRACT_ADDRESS=0x0000000000000000000000000000000000000000
```
Comment (backmeupplz) [inline .env.sample:5]: Let's use `VITE_SC_ERC721_LEDGER_CONTRACT_ADDRESS` here

**Review by backmeupplz — APPROVED** [review]

File: src/components/badges/ListTitle.tsx [hunk 10]
```diff
@@ -17,9 +17,7 @@ function ListTitleSuspended() {
 
   const hasUnminted = proofsAvailableToMint.length > 0
 
-  return !Object.keys(derivativeContracts).length ? (
```
[the remark this answers is not in the document]
Comment (MixailE) [reply src/components/badges/ListTitle.tsx:20]: Yes, if there are no contracts, then infinite loading is shown
